sway/input/seat.c

```diff
@@ -353,68 +354,31 @@ static void handle_new_node(struct wl_listener *listener, void *data) {
 	seat_node_from_node(seat, node);
 }
 
-static void drag_icon_damage_whole(struct sway_drag_icon *icon) {
-	if (!icon->wlr_drag_icon->surface->mapped) {
-		return;
-	}
-	desktop_damage_surface(icon->wlr_drag_icon->surface, icon->x, icon->y, true);
-}
-
-void drag_icon_update_position(struct sway_drag_icon *icon) {
-	drag_icon_damage_whole(icon);
-
-	struct wlr_drag_icon *wlr_icon = icon->wlr_drag_icon;
-	struct sway_seat *seat = icon->seat;
+static void drag_icon_update_position(struct sway_seat *seat, struct wlr_scene_node *node) {
+	struct wlr_drag_icon *wlr_icon = scene_descriptor_try_get(node, SWAY_SCENE_DESC_DRAG_ICON);
```

Comment:
I don't think it's worth it to add a new function for one or maybe two locations that just end up de referencing a NULL pointing anyway.

Reply:
I think it's probably worth it just for the clarity, in general code that looks like

```c
x = try_something_that_might_fail()
use_unconditionally(x)
```

is bound to raise some eyebrows of future readers of the code. I'm fine with not adding a separate function so long as we explicitly assert that we get an output here rather than implicitly relying on crashing on the null deref.